src/IO/H5/Dat.cpp

```diff
@@ -102,7 +102,25 @@ Dat::Dat(const bool exists, detail::OpenGroup&& group, const hid_t location,
   }
 }
 
-Dat::~Dat() { CHECK_H5(H5Dclose(dataset_id_), "Failed to close dataset"); }
+Dat::~Dat() {
+// The pragmas here are used to suppress the warning that the compile time
+// branch of `ERROR` will always call `terminate()` because it throws an error.
+// Throwing that error is a code path that will never actually be entered at
+// runtime, so we suppress the warning here.
+#pragma GCC diagnostic push
+// Suppresses -Wpragmas warning for gcc builds that do not recognize
+// -Wunknown-warning-option
+#pragma GCC diagnostic ignored "-Wpragmas"
+// Suppresses -Wunknown-warning-option for clang builds that do not recognize
+// -Wterminate
+#pragma GCC diagnostic ignored "-Wunknown-warning-option"
```

Comment:
I can do that! The only reason I did it this way is because there are different warning names to suppress for gcc-9/10 and gcc-11. I can look up how to detect which gcc version we have and update it